### PR TITLE
Survival type table 

### PR DESCRIPTION
**Pull request by adamabeshouse**
Fixes https://github.com/cBioPortal/cbioportal/issues/7805

image

[Link1](https://www.cbioportal.org/results/comparison?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=brca_tcga_pan_can_atlas_2018&case_set_id=brca_tcga_pan_can_atlas_2018_cnaseq&comparison_subtab=survival&data_priority=0&gene_list=TP53&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=brca_tcga_pan_can_atlas_2018_gistic&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=brca_tcga_pan_can_atlas_2018_mutations&profileFilter=0&tab_index=tab_visualize)
[Link2](https://genie-private.cbioportal.org/results/comparison?Action=Submit&cancer_study_list=nsclc_genie_bpc&case_set_id=nsclc_genie_bpc_all&gene_list=TP53&tab_index=tab_visualize&comparison_subtab=survival)

run `localStorage.netlify = "deploy-preview-3409--cbioportalfrontend"`

### CODE REVIEW

File: src/pages/resultsView/survival/SurvivalChart.tsx [hunk 1]
```diff
@@ -735,7 +711,9 @@ export default class SurvivalChart
     }
 
     public render() {
-        if (this.props.patientSurvivals.length === 0) {
+        if (
+            _.flatten(_.values(this.props.sortedGroupedSurvivals)).length === 0
```
Comment (kalletlak) [inline src/pages/resultsView/survival/SurvivalChart.tsx:715]: this could be `_.flatMap`

**Review by kalletlak — COMMENTED** [review]
LGTM

File: src/pages/resultsView/survival/SurvivalChart.tsx (hunk 1 above)
Comment (adamabeshouse) [reply src/pages/resultsView/survival/SurvivalChart.tsx:715]: `_.flatMap(survivals, x=>x)`?

**Comment by jjgao** [comment]
@adamabeshouse there was some error when I try to test the second link using localstorage trick. The first link works.

image

**Comment by jjgao** [comment]
@schultzn could you test [this](https://deploy-preview-3409--cbioportalfrontend.netlify.app/results/comparison?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=brca_tcga_pan_can_atlas_2018&case_set_id=brca_tcga_pan_can_atlas_2018_cnaseq&comparison_subtab=survival&data_priority=0&gene_list=TP53&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=brca_tcga_pan_can_atlas_2018_gistic&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=brca_tcga_pan_can_atlas_2018_mutations&profileFilter=0&tab_index=tab_visualize)?

**Comment by adamabeshouse** [comment]
@jjgao oops sorry, should be fixed now

**Comment by jjgao** [comment]
Thanks, @adamabeshouse. It's working now.

**Comment by schultzn** [comment]
Looks great!

Will you have the option to hide the p- and q-values in individual portal instances?

**Comment by adamabeshouse** [comment]
@schultzn thanks! yes, it's controlled by portal.properties https://github.com/cBioPortal/cbioportal/pull/7913/files#diff-d3c4f711d0e1bac0f6b721ed5eb4bfd4R203

**Comment by jjgao** [comment]
@inodb @alisman could you update the genie-private property to exclude p/q when this is released?

**Comment by inodb** [comment]
@jjgao Thanks for reminder - looks like it's been updated: https://github.com/knowledgesystems/knowledgesystems-k8s-deployment/blob/62dcb6861454f15dbd0e2367a582329cbb37bbfd/cbioportal/cbioportal_backend_genie_private.yaml#L143

**Pull request merged**